File: Java/Queue/Python/queue.py
class Queue:
    def __init__(self,size):
        self.head = 1
        self.tail = 1
        self.Q = [0]*(size)
        self.size = size
    
    def is_empty(self):
        if self.tail == self.head:
            return True
        return False

    def is_full(self):
        if self.head == self.tail + 1:
            return True
        return False
    

    def enqueue(self,x):
        if self.is_full():
            print("Queue Overflow")
        else:
            self.Q[self.tail] = x
            if self.tail == self.size:
                self.tail = 1

            else:
                self.tail = self.tail + 1

    def dequeue(self):
        if self.is_empty():
            print("Queue Underflow")
        else:
            x = self.Q[self.head]
            if self.head == self.size:
                self.head = 1
            else:
                self.head +=1
            return x

    def display(self):
        i = self.head
        while(i<self.tail):
            print(self.Q[i])
            if i == self.size:
                i = 0
            i = i +1


if __name__ == "__main__":
    q = Queue(10)
    q.enqueue(10)
    q.enqueue(20)
    q.enqueue(30)
    q.enqueue(40)
    q.enqueue(50)
    q.display()

    print("")

    q.dequeue()
    q.dequeue()
    q.display()

    print("")

    q.enqueue(60)
    q.enqueue(70)
    q.display()

File: Java/Queue/Java/Queue.java
class Queue {
    public int head;
    public int tail;
    public int size;
    public int[] Q;

    public Queue(int size){
        this.head = 1;
        this.tail = 1;
        this.Q = new int[size];
    }

    public boolean isEmpty(){
        if(this.tail == this.head)
        return true;
        return false;
    }

    public boolean isFull(){
        if(this.head == this.tail + 1)
        return true;
        return false;
    }

    public void enqueue(int x){
        if(isFull()){
            System.out.println("Queue OVerflow");
        }
        else{
            this.Q[this.tail] = x;
            if(this.tail == this.size)
            this.tail = 1;
            else
            this.tail = this.tail + 1;
        }
    }

    public int dequeue(){
        if(isEmpty()){
            System.out.println("Underflow");
            return -1000;
        }
        else {
            int x = this.Q[this.head];
            if(this.head == this.size){
                this.head = 1;
            }
            else{
                this.head = this.head + 1;
            }
            return x;
        }
    }

    public void display(){
        int i;
        for (i=this.head;i<this.tail;i++){
            System.out.println(this.Q[i]);
            if(i == this.size){
                i = 0;
            }
        }
    }

    public static void main(String[] args) {
        Queue q = new Queue(10);
    q.enqueue(10);
    q.enqueue(20);
    q.enqueue(30);
    q.enqueue(40);
    q.enqueue(50);
    q.display();

    System.out.println("");

    q.dequeue();
    q.dequeue();
    q.display();

    System.out.println("");

    q.enqueue(60);
    q.enqueue(70);
    q.display();
    }
}